bindings/python/tests/bindings/test_normalizers.py

```diff
@@ -93,3 +94,41 @@ def test_full_strip(self):
 
         output = normalizer.normalize_str("  hello  ")
         assert output == "hello"
+
+
+class TestCustomNormalizer:
+    class BadCustomNormalizer:
+        def normalize(self, normalized, wrong):
+            pass
+
+    class GoodCustomNormalizer:
+        def normalize(self, normalized):
+            self.kept_normalized = normalized
+            normalized.replace("there", "you")
```

Comment:
Careful with this, it doesn't respect the `replace` of Python which is *not* inplace, but returns a new string.

It will be very confusing for users.

Reply:
Sure, but this is really hard to achieve in Rust without a lot of unnecessary unsafe code, or without cloning. We'll have to be clear about this in our guides/documentation.